src/Service/ShipmentService.php

```diff
@@ -347,6 +364,15 @@ public function saveShipment(Order $order, ShipmentData $shipmentData, $print =
             $deliveryTime->update();
         }
 
+        if ($shipmentData->getProduct()) {
+            $product = new DPDProduct($shipmentData->getProduct());
+            $productCarrier = Carrier::getCarrierByReference($product->id_reference);
```

Comment:
remember, avoid static global methods like this one, you can create repository CarrierRepository and hide this functionality in there